test/Common.FunctionalTests/Inprocess/FrebTests.cs

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using Microsoft.AspNetCore.Server.IIS.FunctionalTests.Utilities;
+using Microsoft.AspNetCore.Server.IntegrationTesting;
+using Microsoft.AspNetCore.Server.IntegrationTesting.IIS;
+using Microsoft.AspNetCore.Testing.xunit;
+using Microsoft.Extensions.Logging;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Server.IISIntegration.FunctionalTests
+{
+    [Collection(PublishedSitesCollection.Name)]
+    public class FrebTests : LogFileTestBase
+    {
+        private readonly PublishedSitesFixture _fixture;
+        public FrebTests(PublishedSitesFixture fixture)
+        {
+            _fixture = fixture;
+        }
+
+        public static Dictionary<string, string[]> FrebChecks = InitFrebChecks();
+        public static IEnumerable<object[]> FrebScenarios => FrebChecks.ToTheoryData();
+
+        public static Dictionary<string, string[]> InitFrebChecks()
+        {
+            var dict = new Dictionary<string, string[]>();
+            dict.Add("Request Start", new string[] { "HelloWorld", "ANCM_INPROC_EXECUTE_REQUEST_START" });
+            dict.Add("Request End", new string[] { "HelloWorld", "ANCM_INPROC_EXECUTE_REQUEST_COMPLETION", "1" });
+            dict.Add("Async Start", new string[] { "HelloWorld", "ANCM_INPROC_ASYNC_COMPLETION_START" });
+            dict.Add("Async End", new string[] { "HelloWorld", "ANCM_INPROC_ASYNC_COMPLETION_COMPLETION", "0" });
+            dict.Add("Managed End", new string[] { "HelloWorld", "ANCM_INPROC_MANAGED_REQUEST_COMPLETION" });
+            dict.Add("Request Failure", new string[] { "Throw", "ANCM_INPROC_ASYNC_COMPLETION_COMPLETION", "2" });
+            return dict;
+        }
+
+        [ConditionalTheory]
+        [RequiresIIS(IISCapability.FailedRequestTracingModule)]
+        [MemberData(nameof(FrebScenarios))]
+        public async Task CheckCommonFrebEvents(string scenario)
```

Comment:
You are deploying 5 times to test 1 scenario ("HelloWorld"). I think it's a bit wasteful.